deepspeed/runtime/swap_tensor/utils.py

```diff
@@ -179,9 +180,10 @@ def __init__(self, num_elems, count, dtype):
         self.count = count
         self.dtype = dtype
         self.all_buffers = [
-            torch.zeros(num_elems,
-                        device='cpu',
-                        dtype=dtype).pin_memory() for _ in range(count)
+            get_accelerator().pin_memory(
```

Comment:
Why does this need to be changed to use the `accelerator.pin_memory()` method?

Reply:
For a CPU tensor t, `t.pin_memory()` would pin this tensor to GPU memory.  In new pytorch version (1.13.1), a device string can be passed to pin_memory to specify current device.  This is how `get_accelerator().pin_memory(tensor)` is implemented for XPU:
https://github.com/intel/intel-extension-for-deepspeed/blob/main/intel_extension_for_deepspeed/xpu_accelerator.py#L178

However, in early version of pytorch (1.8.1), pin_memory does not accept device string as named argument, so in CUDA implementation we use the no parameter form to keep pytorch compatibility.
https://github.com/microsoft/DeepSpeed/blob/master/accelerator/cuda_accelerator.py#L215

So we introduced an interface `pin_memory()` in Accelerator for compatibility.  For CUDA device, we will call `pin_memory()` without device parameter so both pytorch 1.8.1 and pytorch 1.13.1 will work.  For XPU device, we will specificy xpu device explicitly to make pin_memory goes to XPU path.

If we run the following with torch 1.13.1 on CUDA device, pin_name with device argument will pass:
```
>>> import torch
>>> a = torch.zeros(3,3)
>>> b = a.pin_memory(device="cuda:0")
>>> a.is_pinned()
False
>>> b.is_pinned()
True
```
With torch 1.8.1 the third line will have an error, indicating pin_memory does not accept device as paramter.

Reply:
thanks for the explanation! I learned something new today :)